File: Structural/Lab8/Typescript/src/program.ts
import { HouseBuilder } from './Builders/house_builder';
import { ApartmentBuilder } from './Builders/apartment_builder';
import { EstateDirector } from './Directors/estate_director';
import { CheapEstateDirector } from './Directors/cheap_estate_director';
import { LuxuryEstateDirector } from './Directors/luxury_estate_director';
import { dir } from 'console';
class Program {
    static main(args: string[]): void {
        console.log("Start testing... \n\n");
        let director : EstateDirector;
        director = new LuxuryEstateDirector(new ApartmentBuilder());
        console.log(`Create new luxury apartment ${director.BuildDefault().toString()}`);
        director = new CheapEstateDirector(new HouseBuilder());
        console.log(`Create new cheap house ${director.BuildDefault().toString()}`);
    }
}

Program.main([]);

File: Structural/Lab8/C#/Program.cs
using System;
using System.Diagnostics;
using System.Net.WebSockets;
using System.Reflection.Emit;
using System.Runtime.InteropServices;
using C_.Builders;
using C_.Directors;
using C_.Entities;

namespace C_
{    
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Start testing... \n\n");
            EstateDirector director;

            director = new LuxuryEstateDirector(new ApartmentBuilder());
            Console.WriteLine($"Create new luxury apartment {director.BuildDefault()}.");
            director = new CheapEstateDirector(new HouseBuilder());
            Console.WriteLine($"Create new cheap house {director.BuildDefault()}.");
        }

    }
}
